SpatialGDK/Source/SpatialGDKTests/SpatialGDK/RPCService/RPCServiceTest.cpp

```diff
@@ -0,0 +1,568 @@
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::Success;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_server_unreliable_rpcs_to_the_service_THEN_check_payloads_are_writen_correctly_to_component_updates)
+{
+	TArray<EntityPayload> EntityPayloads;
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_1, SimplePayload));
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_2, SimplePayload));
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_3, SimplePayload));
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_4, SimplePayload));
+	EntityPayloads.Add(EntityPayload(RPCTestEntityId_5, SimplePayload));
+
+	TArray<Worker_EntityId> EntityIdArray;
+	EntityIdArray.Add(RPCTestEntityId_1);
+	EntityIdArray.Add(RPCTestEntityId_2);
+	EntityIdArray.Add(RPCTestEntityId_3);
+	EntityIdArray.Add(RPCTestEntityId_4);
+	EntityIdArray.Add(RPCTestEntityId_5);
+
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService(EntityIdArray, CLIENT_AUTH);
+	for (const EntityPayload& EntityPayloadItem : EntityPayloads)
+	{
+		RPCService.PushRPC(EntityPayloadItem.EntityId, ERPCType::ServerUnreliable, SimplePayload);
+	}
+
+	TArray<SpatialGDK::SpatialRPCService::UpdateToSend> UpdateToSendArray = RPCService.GetRPCsAndAcksToSend();
+
+	bool bTestPassed = true;
+	if (UpdateToSendArray.Num() != EntityPayloads.Num())
+	{
+		bTestPassed = false;
+	}
+	else
+	{
+		for (int i = 0; i < EntityPayloads.Num(); ++i)
+		{
+			if (!CompareUpdateToSendAndEntityPayload(UpdateToSendArray[i], EntityPayloads[i], ERPCType::ServerUnreliable, 1))
+			{
+				bTestPassed = false;
+				break;
+			}
+		}
+	}
+
+	TestTrue("UpdateToSend have expected payloads", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_no_authority_over_rpc_endpoint_WHEN_push_client_reliable_rpcs_to_the_service_THEN_check_component_data)
+{
+	// Create RPCService with empty component view
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({}, INVALID);
+
+	RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientReliable, SimplePayload);
+
+	Worker_ComponentData ComponentData = GetComponentDataOnEntityCreationFromRPCService(RPCService, RPCTestEntityId_1, ERPCType::ClientReliable);
+	bool bTestPassed = CompareComponentDataAndEntityPayload(ComponentData, EntityPayload(RPCTestEntityId_1, SimplePayload), ERPCType::ClientReliable, 1);
+	TestTrue("Entity creation test returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_no_authority_over_rpc_endpoint_WHEN_push_multicast_rpcs_to_the_service_THEN_check_component_data)
+{
+	// Create RPCService with empty component view
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({}, INVALID);
+
+	const static int RPCsToSend = 5;
+	for (int i = 0; i < RPCsToSend; ++i)
+	{
+		RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	}
```

Comment:
Got a for loop here for 5 things, but then in later tests there are 5 repeated calls not in a for loop, maybe expand this one out?

Reply:
Done